src/Workspaces/CSharp/Portable/Simplification/CSharpSimplificationService.Expander.cs

```diff
@@ -254,9 +255,27 @@ public override SyntaxNode VisitSimpleLambdaExpression(SimpleLambdaExpressionSyn
             public override SyntaxNode VisitArgument(ArgumentSyntax node)
             {
                 _cancellationToken.ThrowIfCancellationRequested();
-
+                string inferredName = null;
+                if (node.Parent.IsKind(SyntaxKind.TupleExpression))
+                {
+                    var tuple = (TupleExpressionSyntax)node.Parent;
+                    var nonDuplicateTupleNames = ExtractTupleNames(tuple);
```

Comment:
(and i actually believe just linearly scanning the tuple will be *faster* in practice).  This ia  case where big-O may say something, but actual practical constants impact you.